io.wcm.caconfig.editor bundle fails to load Sling Initial Content on AEM LTS SP2 (jcr.contentloader 2.6.0 - strict Jakarta JSON parser rejects // comments)

## Environment
- AEM: LTS SP2
- Java: 21
- io.wcm.caconfig.editor version: 1.16.8
- org.apache.sling.jcr.contentloader version: 2.6.0 (bundled with AEM LTS SP2)

## Problem

After upgrading AEM LTS SP2, the `io.wcm.caconfig.editor` bundle fails 
to load its Sling Initial Content, causing all caconfig editor components to be missing from the 
JCR repository.

The error in `error.log` is:
org.apache.sling.jcr.contentloader.internal.BundleContentLoader
Cannot load initial content for bundle io.wcm.caconfig.editor : Unexpected char 47
Caused by: jakarta.json.stream.JsonParsingException: Unexpected char 47 (= '/')
at org.glassfish.json.JsonTokenizer [org.glassfish.jakarta.json:2.0.1]


This means `/apps/wcm-io/caconfig/editor/components/page/editor` (and potentially other nodes) 
are never created in the repository. Any page using the caconfig editor then fails with:

No renderer for extension html, type=wcm-io/caconfig/editor/components/page/editor, superType=null

## Major Cause

AEM LTS SP2 upgraded `org.apache.sling.jcr.contentloader` to **2.6.0**.
This was working in LTS SP1 where org.apache.sling.jcr.contentloader version was 2.5.0

## Steps to Reproduce
1. Install AEM 6.5 LTS SP2 (fresh install)
2. Deploy `io.wcm.caconfig.editor:1.16.8` bundle to the instance
3. Check `error.log` — `JsonParsingException: Unexpected char 47` is logged
4. Navigate to `/apps/wcm-io/caconfig/editor/components/page/editor` in CRXDE — node does not exist


## Expected Behavior
The Sling Initial Content in `SLING-INF/` loads successfully and all component nodes are created 
under `/apps/wcm-io/caconfig/editor/`.

@stefanseifert Thanks for the update
I tried the workaround you provided but i still get the same error while io.wcm.caconfig.editor-1.16.8.jar bundle gets deployed in OSGI
I have both 2.0.0 and 1.2.16 org.apache.sling.commons.johnzon bundle versions active in my OSGI

Can wcm-io provide a hotfix for this or do we have to wait for Adobe team to get the fix

Below are the error logs:

23.02.2026 17:47:25.722 *ERROR* [OsgiInstallerImpl] org.apache.sling.jcr.contentloader.internal.BundleContentLoader Cannot load initial content for bundle io.wcm.caconfig.editor : Unexpected char 47 at (line no=5, column no=3, offset=101)
javax.jcr.RepositoryException: Unexpected char 47 at (line no=5, column no=3, offset=101)
	at org.apache.sling.jcr.contentloader.internal.BundleContentLoader.createNode(BundleContentLoader.java:558) [org.apache.sling.jcr.contentloader:2.6.0]
	at org.apache.sling.jcr.contentloader.internal.BundleContentLoader.installFromPath(BundleContentLoader.java:419) [org.apache.sling.jcr.contentloader:2.6.0]
	at org.apache.sling.jcr.contentloader.internal.BundleContentLoader.installFromPath(BundleContentLoader.java:428) [org.apache.sling.jcr.contentloader:2.6.0]
	at org.apache.sling.jcr.contentloader.internal.BundleContentLoader.installContent(BundleContentLoader.java:298) [org.apache.sling.jcr.contentloader:2.6.0]
	at org.apache.sling.jcr.contentloader.internal.BundleContentLoader.registerBundleInternal(BundleContentLoader.java:185) [org.apache.sling.jcr.contentloader:2.6.0]
	at org.apache.sling.jcr.contentloader.internal.BundleContentLoader.registerBundle(BundleContentLoader.java:141) [org.apache.sling.jcr.contentloader:2.6.0]
	at org.apache.sling.jcr.contentloader.internal.BundleContentLoaderListener.bundleChanged(BundleContentLoaderListener.java:189) [org.apache.sling.jcr.contentloader:2.6.0]
	at org.apache.felix.framework.EventDispatcher.invokeBundleListenerCallback(EventDispatcher.java:915)
	at org.apache.felix.framework.EventDispatcher.fireEventImmediately(EventDispatcher.java:834)
	at org.apache.felix.framework.EventDispatcher.fireBundleEvent(EventDispatcher.java:516)
	at org.apache.felix.framework.Felix.fireBundleEvent(Felix.java:4847)
	at org.apache.felix.framework.StatefulResolver.fireResolvedEvents(StatefulResolver.java:1310)
	at org.apache.felix.framework.StatefulResolver.resolve(StatefulResolver.java:522)
	at org.apache.felix.framework.Felix.resolveBundleRevision(Felix.java:4393)
	at org.apache.felix.framework.Felix.startBundle(Felix.java:2308)
	at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:1006)
	at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:992)
	at org.apache.sling.installer.core.impl.tasks.BundleStartTask.execute(BundleStartTask.java:97) [org.apache.sling.installer.core:3.14.4]
	at org.apache.sling.installer.core.impl.OsgiInstallerImpl.doExecuteTasks(OsgiInstallerImpl.java:926) [org.apache.sling.installer.core:3.14.4]
	at org.apache.sling.installer.core.impl.OsgiInstallerImpl.executeTasks(OsgiInstallerImpl.java:758) [org.apache.sling.installer.core:3.14.4]
	at org.apache.sling.installer.core.impl.OsgiInstallerImpl.run(OsgiInstallerImpl.java:306) [org.apache.sling.installer.core:3.14.4]
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.io.IOException: Unexpected char 47 at (line no=5, column no=3, offset=101)
	at org.apache.sling.jcr.contentloader.internal.readers.JsonReader.parse(JsonReader.java:185) [org.apache.sling.jcr.contentloader:2.6.0]
	at org.apache.sling.jcr.contentloader.internal.readers.JsonReader.parse(JsonReader.java:161) [org.apache.sling.jcr.contentloader:2.6.0]
	at org.apache.sling.jcr.contentloader.internal.BundleContentLoader.createNode(BundleContentLoader.java:552) [org.apache.sling.jcr.contentloader:2.6.0]
	... 21 common frames omitted
Caused by: jakarta.json.stream.JsonParsingException: Unexpected char 47 at (line no=5, column no=3, offset=101)
	at org.glassfish.json.JsonTokenizer.unexpectedChar(JsonTokenizer.java:577) [org.glassfish.jakarta.json:2.0.1]
	at org.glassfish.json.JsonTokenizer.nextToken(JsonTokenizer.java:394) [org.glassfish.jakarta.json:2.0.1]
	at org.glassfish.json.JsonParserImpl$ObjectContext.getNextEvent(JsonParserImpl.java:499) [org.glassfish.jakarta.json:2.0.1]
	at org.glassfish.json.JsonParserImpl.next(JsonParserImpl.java:369) [org.glassfish.jakarta.json:2.0.1]
	at org.glassfish.json.JsonParserImpl.getObject(JsonParserImpl.java:328) [org.glassfish.jakarta.json:2.0.1]
	at org.glassfish.json.JsonParserImpl.getObject(JsonParserImpl.java:166) [org.glassfish.jakarta.json:2.0.1]
	at org.glassfish.json.JsonReaderImpl.readObject(JsonReaderImpl.java:100) [org.glassfish.jakarta.json:2.0.1]
	at org.apache.sling.jcr.contentloader.internal.readers.JsonReader.parse(JsonReader.java:181) [org.apache.sling.jcr.contentloader:2.6.0]
	... 23 common frames omitted

--- a/changes.xml
+++ b/changes.xml
@@ -25,7 +25,7 @@
   <body>
 
     <release version="1.16.10" date="not released">
-      <action type="fix" dev="sseifert">
+      <action type="fix" dev="sseifert" issue="73">
         Strip comments from JSON Files in Sling-Initial-Content to fix deployment problem in AEM 6.6.2 (AEM 6.5 LTS SP 2).
       </action>
     </release>

--- a/changes.xml
+++ b/changes.xml
@@ -24,6 +24,12 @@
     xsi:schemaLocation="http://maven.apache.org/changes/2.0.0 https://maven.apache.org/xsd/changes-2.0.0.xsd">
   <body>
 
+    <release version="1.16.10" date="not released">
+      <action type="fix" dev="sseifert">
+        Strip comments from JSON Files in Sling-Initial-Content to fix deployment problem in AEM 6.6.2 (AEM 6.5 LTS SP 2).
+      </action>
+    </release>
+
     <release version="1.16.8" date="2025-12-08">
       <action type="update" dev="sseifert" issue="66">
         Enable validation for pathbrowser widget.

--- a/parent/pom.xml
+++ b/parent/pom.xml
@@ -25,7 +25,7 @@
   <parent>
     <groupId>io.wcm</groupId>
     <artifactId>io.wcm.parent_toplevel</artifactId>
-    <version>2.6.2</version>
+    <version>2.7.0-SNAPSHOT</version>
     <relativePath />
   </parent>
 

